### PR TITLE
[10.0][ADD][MIG][BR001409] [T27548] auth_cas migrating from v8 to v10

### PR DESCRIPTION
**Pull request by YogeshMahera-SerpentCS**
- auth_cas

### CODE REVIEW

File: auth_cas/README.rst [hunk 1]
```diff
@@ -0,0 +1,80 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: https://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+==================
+CAS Authentication
+==================
+
+This module allows users to login with their CAS account (often LDAP username
+and password), and will automatically create OpenERP users for them on the fly.
```
Comment (elicoidal) [inline auth_cas/README.rst]: Odoo

File: auth_cas/README.rst [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+For use it, you must have a functional CAS server already installed.
```
Comment (elicoidal) [inline auth_cas/README.rst]: In order to use it

File: auth_cas/README.rst [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+No specific steps required
+
+
+Configuration
+=============
+
+After installing this module, you need to configure the CAS parameters in the
+default settings.
+The CAS authentication only works if you are in a single database mode.
+You can launch the OpenERP Server with the option --db-filter=YOUR_DATABASE to do so.
```
Comment (elicoidal) [inline auth_cas/README.rst]: Odoo

File: auth_cas/README.rst [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+Installation
+============
+
+No specific steps required
+
+
+Configuration
+=============
+
+After installing this module, you need to configure the CAS parameters in the
```
Comment (elicoidal) [inline auth_cas/README.rst]: Create a list

File: auth_cas/README.rst [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+Users' CAS passwords are never stored in the OpenERP database, the CAS server
+is queried whenever a user needs to be authenticated.
+OpenERP does not manage password changes in the CAS, so any change of password
+should be conducted by other means.
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/
+project/issues>`_.
```
Comment (elicoidal) [inline auth_cas/README.rst]: odoo-addons

File: auth_cas/__init__.py [hunk 6]
```diff
@@ -0,0 +1,9 @@
+# -*- coding: utf-8 -*-
+# © 2016 Roméo Guillot Roméo Guillot (http://www.opensource-elanz.fr).
+# © 2016 Elico Corp (https://www.elico-corp.com).
```
Comment (elicoidal) [inline auth_cas/__init__.py]: no copyright/license in inti.py

File: auth_cas/__manifest__.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+# © 2016 Roméo Guillot Roméo Guillot (http://www.opensource-elanz.fr).
+# © 2016 Elico Corp (https://www.elico-corp.com).
```
Comment (elicoidal) [inline auth_cas/__manifest__.py]: 2016-2018

File: auth_cas/__manifest__.py [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+{
+    'name': 'CAS Authentication',
+    'version': '8.0.1.0.0',
```
Comment (elicoidal) [inline auth_cas/__manifest__.py]: 10

File: auth_cas/controllers/main.py [hunk 9]
```diff
@@ -0,0 +1,294 @@
...
+_logger = logging.getLogger(__name__)
+
+
+class Controller(http.Controller):
+    _cp_path = '/auth_cas'
+
+    @staticmethod
+    def get_config_static(req, dbname):
+        """ Retrieves the module config for the CAS authentication. """
+        # registry = RegistryManager.get(dbname)
```
Comment (elicoidal) [inline auth_cas/controllers/main.py]: remove dead code

**Review by elicoidal — COMMENTED** [review]
very light review

**Comment by YogeshMahera-SerpentCS** [comment]
@elicoidal @victormartinelicocorp Could you please review it?

File: auth_cas/README.rst [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+
+This module allows users to login with their CAS account (often LDAP username
+and password), and will automatically create Odoo users for them on the fly.
+In Order to use it, you must have a functional CAS server already installed.
+
+
+Installation
+============
+
+* Add CAS server address under Settings/General Settings
```
Comment (elicoidal) [inline auth_cas/README.rst]: better use `#.`
Try to improve the wording.

File: auth_cas/__manifest__.py [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+    'depends': [
+        'base_setup',
+        'web'
+    ],
+    'data': [
+        'wizard/res_config_view.xml',
+        'views/auth_cas_view.xml'
+    ],
+    'images': [
+        'images/cas_auth_settings.jpeg',
```
Comment (elicoidal) [inline auth_cas/__manifest__.py]: Include this images in the README file

File: auth_cas/__manifest__.py [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+        'images/cas_auth_example_login_page.jpeg'
+    ],
+    'auto_install': False,
```
Comment (elicoidal) [inline auth_cas/__manifest__.py]: no need this key

File: auth_cas/controllers/main.py [hunk 13]
```diff
@@ -0,0 +1,183 @@
...
+from odoo.http import request
+
+import werkzeug
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+class Controller(http.Controller):
+    _cp_path = '/auth_cas'
```
Comment (elicoidal) [inline auth_cas/controllers/main.py]: this variable is unused

File: auth_cas/controllers/main.py [hunk 14]
```diff
@@ -0,0 +1,183 @@
...
+
+        def qs(url):
+            query = urlparse.urlparse(url).query
+            res = dict(
+                [(k, v[0]) for k, v in urlparse.parse_qs(query).items()])
+            res1 = {}
+            if res.get('redirect', {}):
+                res1 = qs(res.get('redirect', {}))
+            res.update(res1)
+            # todo if one ticket in redirect
```
Comment (elicoidal) [inline auth_cas/controllers/main.py]: todo -> TODO

File: auth_cas/i18n/en_US.po [hunk 15]
```diff
@@ -0,0 +1,82 @@
+# Translation of Odoo Server.
```
Comment (elicoidal) [inline auth_cas/i18n/en_US.po:1]: add a translation file for Chinese

File: auth_cas/tests/test_auth_cas.py [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+
+from odoo.tests import common
+from mock import patch
+from ..controllers.main import Controller
+
+
+class TestAuthCas(common.TransactionCase):
+
+    def setUp(self):
+        """Create a packaging and wac."""
```
Comment (elicoidal) [inline auth_cas/tests/test_auth_cas.py]: looks a strange docstring for this project

File: auth_cas/tests/test_auth_cas.py [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+        super(TestAuthCas, self).setUp()
+
+    @patch('odoo.addons.auth_cas.controllers.main.request')
+    @patch('odoo.http.request')
+    def test_auth_cas(self, request, request2):
+        """Move stages for Tasks and Issues."""
```
Comment (elicoidal) [inline auth_cas/tests/test_auth_cas.py]: this too is strange

File: auth_cas/views/auth_cas_view.xml [hunk 18]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
```
Comment (elicoidal) [inline auth_cas/views/auth_cas_view.xml]: add copyright

File: auth_cas/wizard/res_config.py [hunk 19]
```diff
@@ -0,0 +1,134 @@
...
+                icp.set_param('cas_auth.cas_server_port', default_port)
+
+        # We save the field used to know
+        # if users have to be created on the fly or not
+        icp.set_param('cas_auth.cas_create_user', str(self.cas_create_user))
+
+    @api.multi
+    def check_cas_server(self):
+
+        """Check if CAS paramaters (host and port) are valids"""
```
Comment (elicoidal) [inline auth_cas/wizard/res_config.py]: valid

Comment (elicoidal) [reply auth_cas/wizard/res_config.py]: if -> whether

File: auth_cas/wizard/res_config.py [hunk 20]
```diff
@@ -0,0 +1,134 @@
...
+        try:
+            # The login function, from pycas,returns 3 if the host is
+            # a CAS host and if parameters given are bad
+            # (Parameters are empty, except the adress of the CAS server,
+            # so there are always bad !)
+            res = login(cas_server, ' ', ' ')
+            if res[0] == 3:
+                title = 'cas_check_success'
+                message = 'Parameters are correct\nThe CAS server is well ' \
+                          'configured !'
```
Comment (elicoidal) [inline auth_cas/wizard/res_config.py]: remove "!"

**Review by elicoidal — CHANGES_REQUESTED** [review]
Please add docstrings.
some details. other than that LGTM.
Add the GIF in the README file and improve the content.

File: auth_cas/tests/test_auth_cas.py (hunk 16 above)
Comment (martintorresvm) [reply auth_cas/tests/test_auth_cas.py]: indeed, please adapt to something simple from Odoo standard

File: auth_cas/wizard/res_config.py [hunk 21]
```diff
@@ -0,0 +1,134 @@
...
+    # Setter is required too
+    @api.multi
+    def set_cas_values(self):
+        self.ensure_one()
+        icp = self.env['ir.config_parameter']
+
+        error = True
+
+        # If the host OR the port is valid
+        if self.cas_server or self.cas_server_port:
```
Comment (martintorresvm) [inline auth_cas/wizard/res_config.py]: What is the point to check if any of them are valid?
Right after is we required that both are valid

File: auth_cas/wizard/res_config.py [hunk 22]
```diff
@@ -0,0 +1,134 @@
...
+            res = login(cas_server, ' ', ' ')
+            if res[0] == 3:
+                title = 'cas_check_success'
+                message = 'Parameters are correct\nThe CAS server is well ' \
+                          'configured !'
+        except Exception, e:
+            _logger.debug(e)
+        # At the moment, I only found this method
+        # in order to show a message after a request
+        raise ValidationError(_(title) + '\n' + _(message))
```
Comment (martintorresvm) [inline auth_cas/wizard/res_config.py]: I rather have a text field with the result of this check than raise a ValidationError 
Would be possible?

**Review by victor-ditay — CHANGES_REQUESTED** [review]
I miss javascript from the original. any reason?

**Comment by YogeshMahera-SerpentCS** [comment]
@victormartinelicocorp 
> I miss javascript from the original. any reason?

I have checked JS code, the JS code functions is not usable,the code is coverd in controllers so we don't need JS file.

File: auth_cas/wizard/res_config.py (hunk 21 above)
Comment (YogeshMahera-SerpentCS) [reply auth_cas/wizard/res_config.py]: yes right I am improving code

File: auth_cas/wizard/res_config.py [hunk 23]
```diff
@@ -126,9 +125,11 @@ def check_cas_server(self):
...
                 message = 'Parameters are correct\nThe CAS server is well ' \
-                          'configured !'
+                          'configured'
         except Exception, e:
             _logger.debug(e)
+            error_message = e
         # At the moment, I only found this method
         # in order to show a message after a request
-        raise ValidationError(_(title) + '\n' + _(message))
+        raise ValidationError(_(title) + '\n' + _(message) + '\n' + _(
```
Comment (martintorresvm) [inline auth_cas/wizard/res_config.py]: is not what I asked for but ok don't lose more time on this.

**Review by victor-ditay — APPROVED** [review]

**Comment by victor-ditay** [comment]
@YogeshMahera-SerpentCS still some comments from @elicoidal not satisfied

**Comment by YogeshMahera-SerpentCS** [comment]
@elicoidal @victormartinelicocorp Could you please review it?

File: auth_cas/models/res_users.py [hunk 24]
```diff
@@ -0,0 +1,41 @@
...
+    in order to check autenticity of logins
+    """
+
+    _inherit = 'res.users'
+
+    cas_key = fields.Char('CAS Key', size=16, readonly=True)
+
+    @api.model
+    def check_credentials(self, password):
+        """Check autenticity of logins"""
```
Comment (elicoidal) [inline auth_cas/models/res_users.py]: Authenticy

File: auth_cas/static/src/css/auth_cas.css [hunk 25]
```diff
@@ -0,0 +1,8 @@
+.cas_check{
+    border-left: 1px solid #DDDDDD;
+    position: relative;
+    display: inline;
+    left: 10px;
+    margin: 10px;
+    top: 15px;
+}
```
Comment (elicoidal) [inline auth_cas/static/src/css/auth_cas.css]: emptty line

**Review by elicoidal — APPROVED** [review]
LGTM but we need to translate the po file before merge

File: auth_cas/static/src/css/auth_cas.css (hunk 25 above)
Comment (YogeshMahera-SerpentCS) [reply auth_cas/static/src/css/auth_cas.css]: Added Empty line

**Comment by YogeshMahera-SerpentCS** [comment]
@elicoidal Could you please check I have translated PO file

**Comment by elicoidal** [comment]
:laughing: thanks!
(although you used traditional Chinese :wink: )
@lonelysun can you have a quick look and correct the translation?

**Review by lonelysun — APPROVED** [review]

**Review by lonelysun — APPROVED** [review]
@elicoidal 
translation is ok.
LGTM

**Pull request merged**